backend/src/channel/channel.service.ts

```diff
@@ -0,0 +1,512 @@
+                    if (channelMember2.MemberType !== 'REQUESTED') {
+                        return 'User is not requested';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'MEMBER');
+                    return 'User accepted';
+                case 'reject':
+                    if (channelMember2.MemberType !== 'REQUESTED') {
+                        return 'User is not requested';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'NOTMEMBER');
+                    return 'User rejected';
+                case 'remove':
+                    if (channelMember2.MemberType === 'NOTMEMBER') {
+                        return 'User is not a member of the channel';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'NOTMEMBER');
+                    return 'User removed';
+                case 'makeAdmin':
+                    if (channelMember.MemberType !== 'OWNER' || channelMember2.MemberType === 'ADMIN') {
+                        return 'Invalid action';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'ADMIN');
+                    return 'User made admin';
+                case 'makeUser':
+                    if (channelMember.MemberType !== 'OWNER' || channelMember2.MemberType !== 'ADMIN') {
+                        return 'Invalid action';
+                    }
+                    await this.updateChannelMember(channelMember2.id, 'MEMBER');
+                    return 'User made user';
+                default:
+                    return 'Invalid action';
+            }
+        }
+        else {
+            return 'Invalid action';
+        }
+    }
+
+    async UserActions(@Req() req: Request, channel_id: number, action: 'join' | 'leave' | 'cancel' | 'accept' | 'reject') {
+        const channel = await prisma.channel.findUnique({
+            where: {
+                id: channel_id,
+            },
+            include: {
+                ChannelMembers: true,
+            },
+        });
+        if (!channel) {
+            return 'Channel not found';
+        }
+        const user = await this.searchUserById(req.user['id']);
+        if (!user) {
+            return 'User not found';
+        }
+        const channelMember = await prisma.channelMembers.findFirst({
+            where: {
+                channel_id: channel_id,
+                user_id: user.id,
+            },
+        });
+        if (!channelMember) {
+            return 'You are not a member of the channel';
+        }
+        if (channelMember.MemberType === 'BANNED') {
+            return 'You are banned from the channel';
+        }
+        if (channelMember.MemberType === 'MUTED') {
+            return 'You are muted in the channel';
+        }
+        if (channelMember.MemberType === 'NOTMEMBER') {
+            return 'You are not a member of the channel';
+        }

```

Comment:
the return statement is misleading use this instead: return 'User is not requesting to join'